From 7d6178c29a9ee32a0f4017e46ac4154c06abdef6 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Mon, 1 May 2023 12:51:19 +0200
Subject: [PATCH] huawei - set to mono sim

In android 13 the support of the two sim cards is not correct. So we disable one modem
---
 rw-system.sh | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index eff61ba..b408070 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -858,7 +858,7 @@ if [ "$has_hostapd" = false ];then
     setprop persist.sys.phh.system_hostapd true
 fi
 
-#Weird /odm/phone.prop Huawei stuff
+# In Huawei phones, the modem configuration is made depending on the model (/odm/phone.prop or /vendor/phone.prop)
 HW_PRODID="$(sed -nE 's/.*productid=([0-9xa-f]*).*/\1/p' /proc/cmdline)"
 [ -z "$HW_PRODID" ] && HW_PRODID="0x$(od -A none -t x1 /sys/firmware/devicetree/base/hisi,modem_id | sed s/' '//g)"
 for part in odm vendor;do
@@ -869,6 +869,21 @@ for part in odm vendor;do
     fi
 done
 
+# In android 13 the support of the two sim cards is not correct. So we disable one
+if getprop ro.vendor.build.fingerprint | grep -iq -E -e 'huawei|honor' || getprop persist.sys.overlay.huawei | grep -iq -E -e 'true'; then
+
+    # Set to mono sim
+    resetprop_phh ro.multi.rild false
+    resetprop_phh persist.radio.multisim.config single
+    resetprop_phh persist.dsds.enabled false
+
+    # ‭1001100110010100‬ = 09994 - Monosim
+    ‭# 1001101010011100‬ = 09A9C - Multisim
+    resetprop_phh persist.radio.modem.cap 09995
+    resetprop_phh ro.config.modem_number 1
+    resetprop_phh ro.config.client_number 1
+fi
+
 # Fix sprd adf for surfaceflinger to start
 # Somehow the names of the device nodes are incorrect on Android 10; fix them by mknod
 if [ -e /dev/sprd-adf-dev ];then
-- 
2.25.1

